spec/unit/rest/client_spec.rb

```diff
@@ -0,0 +1,80 @@
+require 'spec_helper'
+
+require 'puppet/rest/client'
+require 'puppet/rest/route'
+
+describe Puppet::Rest::Client do
+  context "when creating a new client" do
+    let(:route) { Puppet::Rest::Route.new(api: "/fake_api/v1",
+                                          srv_service: :fakeservice,
+                                          default_server: "myserver.com",
+                                          default_port: 555) }
+    let(:ssl_store) { mock 'store' }
+    let(:http) { stub_everything('http', :request_filter => []) }
+    let(:server_resolver) { mock 'resolver', :select_server_and_port => [ route.default_server, route.default_port ] }
+
+    it "configures a base URL based on the provided route" do
+      url = "https://myserver.com:555/fake_api/v1"
+      http.expects(:base_url=).with(url)
+      client = Puppet::Rest::Client.new(route,
+                                        client: http,
+                                        ssl_store: ssl_store,
+                                        server_resolver: server_resolver)
+      expect(client.base_url).to eq(url)
+    end
+
+    it "initializes itself with basic defaults" do
```

Comment:
Is it possible to introspect the third party implementation to see its configuration vs setting expectations on the methods that we call? ie
```ruby
  client = Puppet::Rest::Client.new(route)
  expect(client.client.base_url).to eq("https://myserver.com:555/fake_api/v1")
```

Reply:
I don't really want to expose the client itself, but I'll replace the default reader with one that returns the third-party value.

Reply:
We can also do an `instance_variable_get`